tests/foreman/cli/test_repository.py

```diff
@@ -374,6 +543,65 @@ def test_positive_update_url(self):
                 self.assertEqual(result['url'], url)
 
     @run_only_on('sat')
+    @tier1
+    def test_negative_update_auth_url_forbidden_characters(self):
+        """Verify that repository URL credentials cannot be updated to contain
+        the forbidden characters
+
+        @Feature: Repository
+
+        @Assert: Repository url not updated
+        """
+        new_repo = self._make_repository()
+        # generate a tuple of auth repos with all valid credentials
+        auth_repos = ()
```

Comment:
Since this is going to be something dynamic, better to use lists. Specially that you are using a comprehension later.